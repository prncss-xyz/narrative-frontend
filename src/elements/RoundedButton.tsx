import { Box } from "./shared";

export function RoundedButton({
  active,
  disabled,
  onClick,
  children,
}: {
  active?: boolean;
  disabled?: boolean;
  onClick?: React.MouseEventHandler;
  children: React.ReactNode;
}) {
  return (
    <Box
      as="button"
      p={1}
      border={"solid"}
      borderColor={active ? undefined : "gray3"}
      color={active ? undefined : "gray1"}
      backgroundColor={active ? "white" : "gray3"}
      borderRadius={3}
      onClick={disabled ? undefined : onClick}
      css={{
        cursor: disabled ? undefined : "pointer",
      }}
    >
      {children}
    </Box>
  );
}
