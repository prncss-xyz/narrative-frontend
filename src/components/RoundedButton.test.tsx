import { fireEvent, render, screen } from "@testing-library/react";
import { RoundedButton } from "./RoundedButton";

describe("RoundedButton", () => {
  describe("not disabled", () => {
    it("should trigger when clicked", () => {
      const spy = vi.fn();
      const { container } = render(
        <RoundedButton onClick={spy}>not disabled</RoundedButton>
      );
      fireEvent.click(screen.getByText("not disabled"));
      expect(spy).toHaveBeenCalledTimes(1);
      expect(container).toMatchSnapshot();
    });
  });
  describe("disabled", () => {
    it("should trigger when clicked", () => {
      const spy = vi.fn();
      const { container } = render(
        <RoundedButton disabled={true} onClick={spy}>
          disabled
        </RoundedButton>
      );
      fireEvent.click(screen.getByText("disabled"));
      expect(spy).toHaveBeenCalledTimes(0);
      expect(container).toMatchSnapshot();
    });
  });
});
