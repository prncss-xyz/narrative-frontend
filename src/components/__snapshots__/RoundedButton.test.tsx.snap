// Vitest Snapshot v1

exports[`RoundedButton > disabled > should trigger when clicked 1`] = `
<div>
  <button
    class="css-1gxxcqd-Box e2i8k6n3"
    color="muted"
    font-size="0"
    height="1"
  >
    disabled
  </button>
</div>
`;

exports[`RoundedButton > not disabled > should trigger when clicked 1`] = `
<div>
  <button
    class="css-z6k8zu-Box e2i8k6n3"
    color="muted"
    cursor="pointer"
    font-size="0"
    height="1"
  >
    not disabled
  </button>
</div>
`;
